backbone/src/main/java/org/researchstack/backbone/step/active/recorder/LocationRecorder.java

```diff
@@ -0,0 +1,153 @@
+import android.location.LocationManager;
+import android.os.Bundle;
+import android.util.Log;
+
+import com.google.gson.JsonObject;
+
+import org.researchstack.backbone.step.Step;
+
+import java.io.File;
+
+/**
+ * Created by TheMDP on 2/17/17.
+ */
+
+/**
+ * Since not all apps using this SDK require this LocationRecorder functionality,
+ * we don't include it in the manifest and hence also suppress missing permissions warnings in the compiler
+ */
+@SuppressWarnings({"MissingPermission"})
+public class LocationRecorder extends JsonArrayDataRecorder implements LocationListener {
+
+    private static final String TAG = LocationRecorder.class.getSimpleName();
+
+    public static final String TIMESTAMP_KEY   = "timestamp";
+    public static final String COORDINATE_KEY  = "coordinate";
+    public static final String LONGITUDE_KEY   = "longitude";
+    public static final String LATITUDE_KEY    = "latitude";
+    public static final String ALTITUDE_KEY    = "altitude";
+    public static final String ACCURACY_KEY    = "accuracy";
+    public static final String COURSE_KEY      = "course";
+    public static final String SPEED_KEY       = "speed";
+
+    private JsonObject jsonObject;
+    private JsonObject coordinateJsonObject;
+
+    private LocationManager locationManager = null;
+    private long minTime;
+    private float minDistance;
+
+    /**
+     * @param minTime per Android doc, minimum time interval between location updates, in milliseconds
+     * @param minDistance per Android doc, minimum distance between location updates, in meters, no minimum if zero
+     * @param identifier the recorder's identifier
+     * @param step the step that contains this recorder
+     * @param outputDirectory the output directory of the file that will be written with location data
+     */
+    LocationRecorder(long minTime, float minDistance, String identifier, Step step, File outputDirectory) {
+        super(identifier, step, outputDirectory);
+        this.minTime = minTime;
+        this.minDistance = minDistance;
+    }
+
+    @Override
+    public void start(Context context) {
+
+        if (locationManager == null) {
+            locationManager = (LocationManager) context.getSystemService(Context.LOCATION_SERVICE);
+        }
+
+        // In Android, you can register for both network and gps location updates
+        // Let's just register for both and log all locations to the data file
+        // with their corresponding accuracy and other data associated
+
+        try {
+            locationManager.requestLocationUpdates(
+                    LocationManager.NETWORK_PROVIDER, minTime, minDistance, this);
+        } catch (java.lang.SecurityException ex) {
+            Log.i(TAG, "fail to request location update, ignore", ex);
+        } catch (IllegalArgumentException ex) {
+            Log.d(TAG, "network provider does not exist, " + ex.getMessage());
+        }
+
+        try {
+            locationManager.requestLocationUpdates(
```

Comment:
Do you really want to request updates from both NETWORK and GPS?  As opposed to trying GPS but fallback to NETWORK or such?